File: 2024/Day 1/R/Part 1.R
PATH <- "./2024/Day 1/input.txt"

# Load the text file
data <- read.table(PATH, header=FALSE, sep=" ", stringsAsFactors = FALSE)

left <- sort(data[,1])
right <- sort(data[,4])

distances <- abs(left - right)
result <- sum(distances)

print(result)

File: 2024/Day 1/Python/Part 1.py
PATH = "./2024/Day 1/input.txt"

with open(PATH, "r") as file:
    left = []
    right = []

    for line in file:
        numbers = line.strip().split()
        left.append(int(numbers[0]))
        right.append(int(numbers[1]))

    left.sort()
    right.sort()             

    distances = 0
    for index in range(len(left)):
        distances += abs(right[index] - left[index])
    print(distances)